src/peergos/shared/user/fs/FileTreeNode.java

```diff
@@ -1019,6 +1042,18 @@ public static FileTreeNode createRoot(TrieNode root) {
                             fut.complete(generateThumbnail(bytes));
                         });
                     }
+                }else if(type == FileType.Video) {
+                    if (network.isJavascript()) {
```

Comment:
This is where we could use the interface, say, VideoThumbnailGenerator and put an instance of it in UserContext

Reply:
The established pattern is to call method on the NativeJSThumbnail class. Return type is string (so can do base64 in gtw translated java), this is in contrast the java method  return byte[]